### ISSUE TITLE
Актуализация "римский цифр"

### ISSUE THREAD

**Issue by Lagodny69** [hippogamesunity/LinguoApp#16]
Прошу тебя удалить весь код связанный с римскими цифрами из ЧЕШСКОГО транслятора, иначе он херит перевод.

**Comment by Lagodny69** [comment]
Также из польского транслятора нужно удалить цифру "I" она конкурирует с буквой I из-за чего она не переводится как И.

**Comment by hippogamesunity** [comment]
Закомментировал замену римских в CZ:
![image](https://user-images.githubusercontent.com/7318172/41027448-c100890a-697f-11e8-8513-58b7f0f384b5.png)

В польском попробуй убрать символ отсюда, может быть это то, что ты хочешь получить:
![image](https://user-images.githubusercontent.com/7318172/41027546-fa9a7432-697f-11e8-9c76-dcbff2e23a2f.png)

**Comment by Lagodny69** [comment]
Думаю что с этим латинским I придётся сделать посложнее. Ведь мы хотим чтобы это всё работало корректно. Нужно чтобы символ I переводился как И. А в случае если он стоит после или перед букв "X" "V" он оставался I. Например I było nas mało w XI wieku - И было нас мало в XI веку... 
мы использовали когда-то вот такое 
private const string LatinConsonants = "AaĄąEeĘęOoÓóUuYyIi";
		private const string LatinVowels = "BbCcChchCzczĆćDdDźdźFfGgHhKkLlŁłMmNnŃńPpRrRzrzSsSzszŚśTtWwZzŹźŻż";

		private static Dictionary<string, string> RegexReplace = new Dictionary<string, string>
		{
            { "[V]JA", "ЪЯ" }, { "[V]ja", "ъя" },
            { "[V]JĄ", "ЪѬ" }, { "[V]ją", "ъѭ" },
            { "[V]JE", "ЪЕ" }, { "[V]je", "ъе" },
            { "[V]JĘ", "ЪѨ" }, { "[V]ję", "ъѩ" },
            { "[V]JO", "ЪЁ" }, { "[V]jo", "ъё" },
            { "[V]JÓ", "ЪÉ" }, { "[V]jó", "ъé" },
            { "[V]JU", "ЪЮ" }, { "[V]ju", "ъю" },
            { "[V]JI", "ЪЇ" }, { "[V]ji", "ъї" },         
        };
может и тут прокатит подобное решение

**Comment by hippogamesunity** [comment]
Приведи пример неправильного перевода. Может в регулярке ошибка.

**Comment by Lagodny69** [comment]
ну простой пример, есть буква I она не переводится в И. 
JA I TY. 
Я I Ты, вместо Я И ТЫ,

**Comment by Lagodny69** [comment]
Вопрос всё ещё актуален

**Comment by Lagodny69** [comment]
при переводе из польской латиницы на польскую кириллицу
Нужно чтобы I не переводилось в И только в случаях II, III, а также перед и после V X. Во всех остальных ситуациях она должна транслироваться в И.

**Comment by hippogamesunity** [comment]
Вижу тут только костыль. Технически
JA I TY
не отличается от 
PETER I BYL TYT

Поэтому с твоим предложением Петр I всегда будет Петр И.
Может решить проблему так, добавив к входному тексту требование, чтобы он был нормализован? Т.е. первое слово с заглавной буквы, остальные буквы - маленькие (кроме имен, разумеется).

Откуда вообще взялся этот CAPSLOCK текст ?

**Comment by hippogamesunity** [comment]
Единственное, что я сейчас улучшил - это добавил требование пробела перед римской цифрой.

**Comment by Lagodny69** [comment]
Смотри, как римская цифра I может быть рядом с X, V, I. Кроме того, в польском языке I это союз в предложении, как и в русском. I był on tu długo - И был он ту длуго. Это не должно быть "I был он ту длуго". Предлагаю забить на это одинокое I. Пусть всегда переводится как И. Однако, когда оно находится после другого и, типа II или после XI то не переводить. Это мне кажется лучшее решение.

**Comment by hippogamesunity** [comment]
Ситуацию, когда предложение начинается с I можно обработать (не считать I цифрой). Давай так и сделаем?
В остальных случаях заглавная I это 100% цифра. По орфографии ну не может заглавная I соединять два слова в середине предложения.

**Comment by Lagodny69** [comment]
так, ты прав. Это неплохое решение, но я бы настоял на том чтобы как цифра она была только с I X V. потому что слово может быть написано капсом, типа PIĆ ПИТЬ

**Comment by hippogamesunity** [comment]
Логично, попробую сделать. Так же и с XV может быть, если аббревиатура, например.

**Comment by Lagodny69** [comment]
Посмотрим что выйдет, можно было бы так и с чешским сделать, но к сожалению в чешском языке используются X V, а в польском нет, это усложнило бы задачу, поэтому мы просто закоментили

**Comment by hippogamesunity** [comment]
Я проверил регулярку - она так и работает, цифры определяет только из диапазона IÎVX.
PIĆ ПИТЬ так и переводится, проблемы не вижу.
